src/screens/QRScreens/ShowPSBT.tsx

```diff
@@ -38,7 +38,20 @@ function ShowPSBT() {
         </ScrollView>
       ) : null}
       <Box style={styles.ctaContainer}>
-        <Buttons primaryText="Done" primaryCallback={() => navigation.pop(2)} />
+        <Buttons
+          primaryText="Done"
+          primaryCallback={() => {
+            navigation.dispatch((state) => {
+              const index = state.routes.findIndex(
+                (route) => route.name === 'SignerAdvanceSettings'
+              );
+              if (index === -1) {
+                return StackActions.replace('SignerAdvanceSettings');
```

Comment:
If you can't find the SignerAdvanceSettings I think you should just do navigation.pop(2) since this is not only used with the RK flow but also normal signing flow